sunpy/map/tests/strategies.py

```diff
@@ -19,10 +17,8 @@ def matrix_meta(draw, key):
         float, (2, 2),
         elements=st.floats(min_value=-1, max_value=1, allow_nan=False))
     )
-    # Make sure matrix isn't singular
-    with warnings.catch_warnings():
-        warnings.filterwarnings(action='ignore', category=RuntimeWarning, module='numpy.linalg')
-        assume(np.abs(np.linalg.det(arr)) > 1e-8)
```

Comment:
We assume somehow the combo of numpy and hypo broke the test on macos?

Reply:
Yes.  I believe it also broke under `ubuntu-latest` once, but succeeded on a later run.  (Are we not caching hypothesis failures in any way?)

Reply:
Ah, I only saw the MacOS fail.

I don't think we are caching anything to do with hypothesis, we could look into it?

Reply:
`hypothesis` naturally caches when run locally, but I don't know how hard it is to set up in the CI.

Reply:
Conner added support for caching items between builds (like our documentation build).